tests/test_computeHolePositions.py

```diff
@@ -0,0 +1,110 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import unittest
+
+from lsst.geom import degrees
+from lsst.cbp import computeHolePositions, MaskInfo
+import lsst.utils.tests
+from lsst.cbp.testUtils import SampleCoordinateConverter
+
+
+class ComputeHolePositionsTestCase(lsst.utils.tests.TestCase):
+
+    def setUp(self):
+        self.scc = SampleCoordinateConverter()
+        self.cco = self.scc.coordinateConverter
+        # point the telescope and CCB right at each other
+        self.cco.setFocalPlanePos(pupilPos=(0, 0))
+
+    def tearDown(self):
+        del self.scc
+        del self.cco
+
+    def testSetup(self):
+        self.assertAnglesAlmostEqual(self.cco.telAzAltInternal.separation(self.cco.cbpAzAltInternal),
+                                     180*degrees)
+        self.assertAnglesAlmostEqual(self.cco.telRotInternal, 0*degrees)
+
+    def testOneDetector(self):
+        detectorPositions = ((100, 250), (250, 100), (500, 1500))
+        for detector in self.cco.cameraGeom:
+            detectorName = detector.getName()
+            holePositions = computeHolePositions(
+                detectorNames=[detectorName],
+                detectorPositions=detectorPositions,
+                cameraGeom=self.cco.cameraGeom,
+                cbpFlipX=self.cco.config.cbpFlipX,
+                cbpFocalLength=self.cco.config.cbpFocalLength,
+            )
+            self.assertEqual(len(holePositions), len(detectorPositions))
+            self.cco.maskInfo = MaskInfo(
+                name="OneDetector",
+                holePositions=holePositions,
+                defaultHole=0,
+            )
+            self.assertEqual(len(self.cco), len(detectorPositions))
+            for i, beamInfo in enumerate(self.cco):
+                self.assertTrue(beamInfo.isOnDetector)
+                self.assertEqual(beamInfo.detectorName, detectorName)
+                self.assertPairsAlmostEqual(beamInfo.detectorPos, detectorPositions[i])
+
+    def testOneHoleAllDetectors(self):
+        detectorPosition = (449.2, 732.1)
+        detectorNames = set(self.cco.cameraGeom.getNameIter())
+        numDetectors = len(self.cco.cameraGeom)
+        for cbpFlipX in (False, True):
+            self.cco.config.cbpFlipX = cbpFlipX
+            holePositions = computeHolePositions(
+                detectorNames=None,
+                detectorPositions=[detectorPosition],
+                cameraGeom=self.cco.cameraGeom,
+                cbpFlipX=self.cco.config.cbpFlipX,
+                cbpFocalLength=self.cco.config.cbpFocalLength,
+            )
+            if not cbpFlipX:
+                flippedHolePositions = [(-val[0], val[1]) for val in holePositions]
+            else:
+                self.assertPairListsAlmostEqual(holePositions, flippedHolePositions)
```

Comment:
My apologies. This is a confusingly written test. What's going on is that cbpFlipX is False the first time through the loop; flippedHolePositions is computed then, and used the next time through the loop when cbpFlipX is True. I have not thought of a cleaner way to do this so I have done two thing:
- added code comments explaining what is going on
- initialize flippedHolePositions to None before the loop, with an explanation. This will make properly fussy linters happy (I'm not sure why flake8 accepted it) and so if things get messed up the equality will fail, rather than giving a mysterious message about an unknown variable

Reply:
That sounds great, thanks 🙂